src/coreclr/src/vm/zapsig.cpp

```diff
@@ -677,16 +684,19 @@ Module *ZapSig::DecodeModuleFromIndexIfLoaded(Module *fromModule,
         pAssembly = fromModule->GetAssembly();
     else
     {
-        if (index < fromModule->GetAssemblyRefMax())
+        DomainCompositeImage *compositeImage = fromModule->GetCompositeImage();
+        uint32_t assemblyRefMax = (compositeImage != NULL ? 1 : fromModule->GetAssemblyRefMax());
```

Comment:
Should this rather get 0 for composite images?

Reply:
Fixed. This requires bits of counterpart changes to Crossgen2 and R2RDump I am adding to the PR in 7th commit.